precompiles/parachain-staking/src/mock.rs

```diff
@@ -0,0 +1,373 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Test utilities
+use super::*;
+use codec::{Decode, Encode};
+use frame_support::{
+	construct_runtime, parameter_types,
+	traits::{GenesisBuild, OnFinalize, OnInitialize},
+	weights::Weight,
+};
+use pallet_evm::{AddressMapping, EnsureAddressNever, EnsureAddressRoot, PrecompileSet};
+use parachain_staking::{InflationInfo, Range};
+use serde::{Deserialize, Serialize};
+use sp_core::{H160, H256};
+use sp_io;
+use sp_runtime::{
+	testing::Header,
+	traits::{BlakeTwo256, IdentityLookup},
+	Perbill, Percent,
+};
+
+pub type AccountId = TestAccount;
+pub type Balance = u128;
+pub type BlockNumber = u64;
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>},
+		Evm: pallet_evm::{Pallet, Call, Storage, Event<T>},
+		Timestamp: pallet_timestamp::{Pallet, Call, Storage, Inherent},
+		ParachainStaking: parachain_staking::{Pallet, Call, Storage, Config<T>, Event<T>},
+	}
+);
+
+// FRom https://github.com/PureStake/moonbeam/pull/518. Merge to common once is merged
```

Comment:
move to precompile-utils

Reply:
someone else can do this in a follow up, I didn't want to add serde, etc as dependencies to `precompile-utils`

Reply:
I agree moving this is beyond the scope of this PR. And honestly, this idea is bigger than just precompiles. We could use this test account for our pallet tests too.